### PR TITLE
Match configuration files wildcard

### LINKED ISSUE #390: Add pattern matching fileName in ANTLRv4GrammarPropertiesStore#findSettingsForFile
currently there is only exact file matching, I would suggest adding matching using Java pattern

### PR DESCRIPTION
**Pull request by wojciszek**
Fixes #390

### CODE REVIEW

File: src/main/java/org/antlr/intellij/plugin/configdialogs/ANTLRv4GrammarPropertiesStore.java [hunk 1]
```diff
@@ -59,14 +60,14 @@ public ANTLRv4GrammarProperties getOrCreateGrammarProperties(String grammarFile)
 	@Nullable
 	private ANTLRv4GrammarProperties findSettingsForFile(String fileName) {
 		for (ANTLRv4GrammarProperties settings : perGrammarGenerationSettings) {
-			if (settings.fileName.equals(fileName)) {
+			WildcardFileNameMatcher wildcardFileNameMatcher = new WildcardFileNameMatcher(settings.fileName);
```
Comment (bintom) [inline src/main/java/org/antlr/intellij/plugin/configdialogs/ANTLRv4GrammarPropertiesStore.java]: Did you check bahavior of invalid pattern given to the constructor?

Comment (wojciszek) [reply src/main/java/org/antlr/intellij/plugin/configdialogs/ANTLRv4GrammarPropertiesStore.java]: I added something like this `"_gdsgjdwwf/\\   dfs"` and no exception has been thrown. It just didn't match any filename input i provided (as expected).

Comment (bjansen) [reply src/main/java/org/antlr/intellij/plugin/configdialogs/ANTLRv4GrammarPropertiesStore.java]: What about invalid regexps like `[` or `*`?

Comment (wojciszek) [reply src/main/java/org/antlr/intellij/plugin/configdialogs/ANTLRv4GrammarPropertiesStore.java]: Special regex characters are escaped. Yet I can try-catch this and log warning just-in-case. What do you think?

**Comment by bjansen** [comment]
Thanks a lot for your contribution, and sorry for the delay :)

**Pull request merged**